sstables/sstables.cc

```diff
@@ -3155,24 +3155,22 @@ delete_atomically(std::vector<shared_sstable> ssts) {
 // since this is an indication we crashed in the middle of delete_atomically
 future<> replay_pending_delete_log(sstring pending_delete_log) {
     sstlog.debug("Reading pending_deletes log file {}", pending_delete_log);
-    return seastar::async([pending_delete_log = std::move(pending_delete_log)] {
         sstring pending_delete_dir = parent_path(pending_delete_log);
         assert(sstable::is_pending_delete_dir(fs::path(pending_delete_dir)));
         try {
             auto sstdir = parent_path(pending_delete_dir);
-            auto text = seastar::util::read_entire_file_contiguous(fs::path(pending_delete_log)).get0();
+            auto text = co_await seastar::util::read_entire_file_contiguous(fs::path(pending_delete_log));
 
             sstring all(text.begin(), text.end());
             std::vector<sstring> basenames;
             boost::split(basenames, all, boost::is_any_of("\n"), boost::token_compress_on);
             auto tocs = boost::copy_range<std::vector<sstring>>(basenames
                     | boost::adaptors::filtered([] (auto&& basename) { return !basename.empty(); })
                     | boost::adaptors::transformed([&sstdir] (auto&& basename) { return sstdir + "/" + basename; }));
-            delete_sstables(tocs).get();
+            co_await delete_sstables(tocs);
         } catch (...) {
```

Comment:
nit: we can later move tocs into delete_sstables()

Reply:
Patch titled 'sstables: Open-code delete_sstables() call' just removes this helper